src/main/java/com/unzer/payment/BasketItem.java

```diff
@@ -20,145 +20,299 @@
  * #L%
  */
 
+import com.google.gson.annotations.SerializedName;
+
 import java.math.BigDecimal;
 import java.net.URL;
 
 public class BasketItem {
-
-	private String basketItemReferenceId;
-	private Integer quantity;
-	private Integer vat;
-	private BigDecimal amountDiscount;
-	private BigDecimal amountGross;
-	private BigDecimal amountVat;
-	private BigDecimal amountPerUnit;
-	private BigDecimal amountNet;
-	private String unit;
 	private String title;
 	private String subTitle;
 	private URL imageUrl;
-	private String type;
+	private String basketItemReferenceId;
+	private Integer quantity;
+	private BigDecimal amountPerUnitGross;
+	private BigDecimal vat;
+	private Type type;
+	private String unit;
+	private BigDecimal amountDiscountPerUnitGross;
 	private String participantId;
-	
+
+	/**
+	 * The title of the basket item
+	 */
+	public String getTitle() {
+		return title;
+	}
+
+	/**
+	 * The title of the basket item
+	 */
+	public BasketItem setTitle(String title) {
+		this.title = title;
+		return this;
+	}
+
+	/**
+	 * The subtitle for the basket item
+	 */
+	public String getSubTitle() {
+		return subTitle;
+	}
+
+	/**
+	 * The subtitle for the basket item
+	 */
+	public BasketItem setSubTitle(String subTitle) {
+		this.subTitle = subTitle;
+		return this;
+	}
+
+	/**
+	 * URL to the basket item image, only used for the payment page.
+	 */
+	public URL getImageUrl() {
+		return imageUrl;
+	}
+
+	/**
+	 * URL to the basket item image, only used for the payment page.
+	 */
+	public BasketItem setImageUrl(URL imageUrl) {
+		this.imageUrl = imageUrl;
+		return this;
+	}
+
+	/**
+	 * Unique basket item reference ID (within the basket).
+	 * If not provided, an ID is automatically generated by Unzer.
+	 */
 	public String getBasketItemReferenceId() {
 		return basketItemReferenceId;
 	}
 
+	/**
+	 * Unique basket item reference ID (within the basket).
+	 * If not provided, an ID is automatically generated by Unzer.
```

Comment:
Description seem to not fit. reference Id is mandatory, and will not be set by Unzer, as far as I know.